mainhandler/vulnscan.go

```diff
@@ -399,17 +427,19 @@ func (actionHandler *ActionHandler) getAuthConfig(pod *corev1.Pod, imageTag stri
 		}
 	}
 
-	// TODO: this should not happen every scan
-	// build a list of secrets from the the registry secrets
-	secrets, err := cloudsupport.GetImageRegistryCredentials(imageTag, pod)
-	if err != nil {
-		return nil, err
-	}
-	for i := range secrets {
-		registryAuth = append(registryAuth, secrets[i])
+	if pod != nil {
```

Comment:
this check is needed go make the code testable (since `GetImageRegistryCredentials` calls `NewKubernetesApi`